### PR TITLE
MM-12355: Add CLI command "command create"

### PR DESCRIPTION
**Pull request by mojicaj**
#### Summary
Implements `command create` to create slash commands in the CLI.

#### Ticket Link
https://github.com/mattermost/mattermost-server/issues/9489

#### Checklist
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

File: cmd/mattermost/commands/command.go [hunk 1]
```diff
@@ -67,3 +89,80 @@ func moveCommandCmdF(command *cobra.Command, args []string) error {
...
+
+	if err := createCommand(a, team, newCommand); err != nil {
+		return errors.New("unable to create command '" + newCommand.Trigger + "'. " + err.Error())
+	}
+	CommandPrettyPrintln("created command '" + newCommand.Trigger + "'")
+
+	return nil
+}
+
+func createCommand(a *app.App, team *model.Team, command *model.Command) *model.AppError {
```
Comment (jespino) [inline cmd/mattermost/commands/command.go]: I don't get the point of this wrapper, we can use directly the `a.CreateCommand`, is there any reason to put it in a separate function?

File: cmd/mattermost/commands/command.go [hunk 2]
```diff
@@ -67,3 +89,80 @@ func moveCommandCmdF(command *cobra.Command, args []string) error {
...
 }
+
+func createCommandCmdF(command *cobra.Command, args []string) error {
+	a, err := InitDBCommandContextCobra(command)
+	if err != nil {
+		return err
+	}
+	defer a.Shutdown()
+
+	if len(args) < 1 {
```
Comment (jespino) [inline cmd/mattermost/commands/command.go]: You can use the `cobra.MinimunNArgs` restriction to do exactly that. An example here: https://github.com/mattermost/mattermost-server/pull/9512/files#diff-6889f01f236fa4b8a3050c7954ec0b68R68

File: cmd/mattermost/commands/command_test.go [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+	"github.com/mattermost/mattermost-server/api4"
+)
+
+func TestCreateCommand(t *testing.T) {
+	th := api4.Setup().InitBasic()
+	defer th.TearDown()
+	team := th.BasicTeam
+	user := th.BasicUser
+
+	CheckCommand(t, "command", "create", team.Name, "--trigger-word", "testCmd", "--url", "http://localhost:8000/my-slash-handler", "--creator", user.Username)
```
[the remark this answers is not in the document]
Comment (mojicaj) [reply cmd/mattermost/commands/command_test.go]: I will work on adding more tests

File: cmd/mattermost/commands/command.go (hunk 1 above)
Comment (mojicaj) [reply cmd/mattermost/commands/command.go]: I wasn't exactly sure either, but I was trying keep conformity with the way the `move` command was setup.  I can certainly remove it.

File: cmd/mattermost/commands/command.go (hunk 2 above)
Comment (mojicaj) [reply cmd/mattermost/commands/command.go]: will do

File: cmd/mattermost/commands/command.go (hunk 1 above)
Comment (jespino) [reply cmd/mattermost/commands/command.go]: Yes, I think in this cases doesn't provide any value and can generate some confusion.

**Comment by mojicaj** [comment]
@jespino how does Mattermost handle resolving conversations.  Do I resolve them as I complete them or does the reviewer do so when they've confirmed it was done?

**Comment by jespino** [comment]
@mojicaj We don't have a policy about that, as a reviewer, I going to check everything before give you the approval anyway, so, feel free to mark things as resolved whenever if that helps you to organize yourself.

File: cmd/mattermost/commands/command.go [hunk 4]
```diff
@@ -67,3 +90,71 @@ func moveCommandCmdF(command *cobra.Command, args []string) error {
...
+	team := getTeamFromTeamArg(a, args[0])
+	if team == nil {
+		return errors.New("unable to find team '" + args[0] + "'")
+	}
+
+	title, _ := command.Flags().GetString("title")
+	description, _ := command.Flags().GetString("description")
+	trigger, errt := command.Flags().GetString("trigger-word")
+	if errt != nil || trigger == "" {
+		return errors.New("atrigger word is required")
```
Comment (jespino) [inline cmd/mattermost/commands/command.go]: You can simplify this kind of errors using the `MarkFlagRequired` for the mandatory flags. more info here: https://github.com/spf13/cobra#required-flags

**Comment by mojicaj** [comment]
Tests added and all changes requested made.  Let me know if there is anything else needed.

**Comment by jespino** [comment]
Sorry, this needs a rebase or merge from master

**Comment by jespino** [comment]
Actually this is a bit weird, you PR includes more commits and changes than are in your commits

**Comment by mojicaj** [comment]
I did do a rebase before I pushed.  I don't know where those commits came from if not from master.
I'll try doing it over and see if that helps.

**Comment by mojicaj** [comment]
Ok, I rebased again and resolved the conflicts.  I see all the same commits that master has.

**Comment by jespino** [comment]
Yes, but you still have a lot of changes not related to your changes in the PR. My proposal would be create a new branch from master, including only your changes, and create a new PR.

**Pull request closed without merging**